preprocessors/object-detection-azure/objdetect.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2021 IMAGE Project, Shared Reality Lab, McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# and our Additional Terms along with this program.
+# If not, see
+# <https://github.com/Shared-Reality-Lab/IMAGE-server/LICENSE>.
+
+import requests  # pip3 install requests
+from re import search
+import operator
+
+# import numpy as np
+import json
+import time
+import jsonschema
+import logging
+import base64
+import os
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+
+
+def process_results(result):
+    send = []
+    image_height = result["metadata"]["height"]
+    image_width = result["metadata"]["width"]
+    for item in range(len(result['objects'])):
+        dims = result["objects"][item]['rectangle']
+        x_left = dims['x'] / image_width
+        x_right = (dims['x'] + dims['w']) / image_width
+        y_left = dims['y'] / image_height
+        y_right = (dims['y'] + dims['h']) / image_height
+        centre = [abs((x_left + x_right) / 2),
+                  abs((y_left + y_right) / 2)]
+        area = abs((dims['w'] * dims['h']) / (image_height * image_width))
+        class_prob = float(result['objects'][item]['confidence'])
+        normalised_dims = [x_left, y_left, x_right, y_right]
+        dictionary = {"ID": item,
+                      "type": result['objects'][item]['object'],
+                      "dimensions": normalised_dims,
+                      "confidence": class_prob,
+                      "centroid": centre,
+                      "area": area
+                      }
+        send.append(dictionary)
+    return send
+
+
+def process_image(image):
+
+    region = "canadacentral"  # For example, "westus"
+    api_key = os.environ["AZURE_API_KEY"]
```

Comment:
Crashes if API key is missing. Should fail gracefully.

Reply:
Same as below, if there's a document with referencing to common error codes, please share.

Reply:
Return 500 if there's no API key and include a log message that there's no key. This just throws a KeyError as is.

Reply:
What does 500 stand for? . There can be many issues with the key, the file can be missing or the key might be outdated. Does this error code handle all such cases ?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500 It's an internal server error. You can specify what happened in the log.